exporter/collector/internal/datapointstorage/datapointcache.go

```diff
@@ -211,28 +211,41 @@ func (c *Cache) gc(shutdown <-chan struct{}, tickerCh <-chan time.Time) bool {
 	return true
 }
 
-// Identifier returns the unique string identifier for a metric.
-func Identifier(resource *monitoredrespb.MonitoredResource, extraLabels map[string]string, metric pmetric.Metric, attributes pcommon.Map) string {
-	var b strings.Builder
-
-	// Resource identifiers
-	if resource != nil {
-		fmt.Fprintf(&b, "%v", resource.GetLabels())
-	}
-
-	// Instrumentation library labels and additional resource labels
-	fmt.Fprintf(&b, " - %v", extraLabels)
+var (
```

Comment:
It might be nice to leave a comment on how you chose these separators. Is there a guarantee that the individual strings wouldn't contain these bytes?

Reply:
added a comment.  They are the delimiters used by prometheus/prometheus for similar hashing.  I added a link above the constants.